public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -146,33 +149,22 @@ export class PanelModel implements DataConfigSource {
 
   constructor(model: any) {
     this.events = new Emitter();
-    // should not be part of defaults as defaults are removed in save model and
-    // this should not be removed in save model as exporter needs to templatize it
-    this.datasource = null;
     this.restoreModel(model);
     this.replaceVariables = this.replaceVariables.bind(this);
   }
 
   /** Given a persistened PanelModel restores property values */
   restoreModel(model: any) {
     // Start with clean-up
-    for (const property of Object.keys(this)) {
-      if (notPersistedProperties[property]) {
-        continue;
-      }
-
-      if (mustKeepProps[property]) {
```

Comment:
Why does the check for mustKeepProps[property] not apply any longer?

Reply:
This clause here was wrong, the mustKeepProps is only what properties should be preserved when switching between panel types. Here what we want to do is basically remove ALL delete all non peristed properties and then re-add properties from the passed in model